### PR TITLE
Install icon & .desktop file to XDG data

### PR DESCRIPTION
**Pull request by luke-jr**
With #13809 (packaging split off), we lost the `.desktop` and `.protocol` files.

This PR restores them, and puts them in their proper places as part of `make install`. (Systems without XDG can be configured `--without-xdg` to avoid this, and the macOS/Windows gitian descriptors are updated to use it.)

Note that:

* `bitcoin128.svg` needs to be named such because (for some reason) `bitcoin128` has become the de facto standard XDG icon name for the generic Bitcoin icon.
* `org.bitcoin.bitcoin-qt.desktop` is required to match `QAPP_ORG_DOMAIN`, which we have set to `"bitcoin.org"` for some reason. (Changing that is out of scope for this PR)

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Build fails with 
```
make[1]: *** No rule to make target 'src/qt/res/src/bitcoin128.svg', needed by 'all-am'.  Stop.

**Comment by hebasto** [comment]
Could the `Comment` field in the `.desktop` file be updated with more languages?

**Review by sharidz313 — APPROVED** [review]

**Comment by luke-jr** [comment]
@MarcoFalke Why? It's part of the git branch...

**Review by sharidz313 — APPROVED** [review]

**Review by sharidz313 — APPROVED** [review]

**Review by sharidz313 — APPROVED** [review]

File: Makefile.am [hunk 1]
```diff
@@ -51,7 +51,7 @@ DIST_SHARE = \
 BIN_CHECKS=$(top_srcdir)/contrib/devtools/symbol-check.py \
            $(top_srcdir)/contrib/devtools/security-check.py
 
-BITCOIN_SVG = src/qt/res/src/bitcoin128.svg
```
Comment (sharidz313) [inline Makefile.am]: Wow

**Review by sharidz313 — APPROVED** [review]

**Comment by laanwj** [comment]
Concept ACK, this the better way to do it IMO than installing it with the packaging.

**Comment by luke-jr** [comment]
@hebasto After this PR, I don't see why not.

**Comment by hebasto** [comment]
Concept ACK.

**Comment by laanwj** [comment]
Regarding the build system issue, it looks like the icon is not part of `make dist`? Likely it needs to be added to another makefile variable.

**Comment by jonasschnelli** [comment]
Concept ACK

> bitcoin128.svg needs to be named such because (for some reason) bitcoin128 has become the de facto standard XDG icon name for the generic Bitcoin icon

Do we want to set the standard Bitcoin icon? IMO Bitcoin Core is a specific application and not sure if that icon should be the default "Bitcoin Icon".

**Comment by luke-jr** [comment]
There we go, Travis is happy now. Shall I squash?

**Comment by luke-jr** [comment]
@jonasschnelli Bitcoin Core doesn't have its own icon at this time - we've always used the generic Bitcoin icon. When/if it does, then the PR adding it can change to use that instead...

**Comment by hebasto** [comment]
The file name `org.bitcoin.bitcoin-qt.desktop` does not follow [Desktop Entry Specification](https://standards.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#file-naming):

> The name of the desktop entry should follow the "reverse DNS" convention: it should start with a reversed DNS domain name controlled by the author of the application, in lower case. The domain name should be followed by the name of the application, which is conventionally written with words run together and initial capital letters (CamelCase). For example, if the owner of `example.org` writes "Foo Viewer", they might choose the name `org.example.FooViewer`, resulting in a file named org.example.FooViewer.desktop.

> Well-known names containing the dash are allowed but not recommended, because the dash is not allowed in some related uses of reversed DNS names, such as D-Bus object paths and interface names, and Flatpak app IDs. If the author's domain name contains a dash, replacing it with an underscore is recommended: this cannot cause ambiguity, because underscores are not allowed in DNS domain names.

**Review by hebasto — CHANGES_REQUESTED** [review]

File: share/qt/org.bitcoin.bitcoin-qt.desktop.in [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+Encoding=UTF-8
+Version=1.0
+Name=@PACKAGE_NAME@
+GenericName=Bitcoin full node@AND_WALLET@
+Comment=Connect to the Bitcoin P2P Network
+Comment[de]=Verbinde mit dem Bitcoin peer-to-peer Netzwerk
+Comment[fr]=Bitcoin, monnaie virtuelle cryptographique pair à pair
+Comment[tr]=Bitcoin, eşten eşe kriptografik sanal para birimi
+TryExec=bitcoin-qt
+Exec=bitcoin-qt %u
```
[the remark this answers is not in the document]
Comment (luke-jr) [reply share/qt/org.bitcoin.bitcoin-qt.desktop.in]: Why not? We expect URIs.

File: share/qt/org.bitcoin.bitcoin-qt.desktop.in [hunk 3]
```diff
@@ -0,0 +1,17 @@
+[Desktop Entry]
```
[the remark this answers is not in the document]
Comment (luke-jr) [reply share/qt/org.bitcoin.bitcoin-qt.desktop.in]: This file predates the PR. It's just being restored at this time.

**Comment by luke-jr** [comment]
@hebasto 
>* `org.bitcoin.bitcoin-qt.desktop` is required to match `QAPP_ORG_DOMAIN`, which we have set to `"bitcoin.org"` for some reason. (Changing that is out of scope for this PR)

**Comment by laanwj** [comment]
> org.bitcoin.bitcoin-qt.desktop is required to match QAPP_ORG_DOMAIN, which we have set to "bitcoin.org" for some reason.

I think what he means is the second part, suggesting replacing `-` with `_`

**Comment by luke-jr** [comment]
Ok, renamed it

**Comment by A6GibKm** [comment]
Consider adding a appdata as in 
https://github.com/MarcoFalke/flathub/blob/f3d313f7991f6fec14f7ddc2c0fec19973f69814/org.bitcoincore.bitcoin-qt.appdata.xml
This should be installed to `{~/.local,/usr/local}/share/metainfo/bitcoin.appdata.xml` (or to `/usr/share...` but `/usr/local` should be prefered to `/usr`)

**Comment by hebasto** [comment]
The alternative approach is https://github.com/bitcoin-core/gui/pull/38 that does not require `make install`, and works with downloaded official binaries.

**Comment by luke-jr** [comment]
Rebased

**Comment by A6GibKm** [comment]
Sorry about that, here is the working [link](https://github.com/flathub/org.bitcoincore.bitcoin-qt/blob/master/org.bitcoincore.bitcoin-qt.metainfo.xml). Which I just realized needs the 0.20.1 release.

**Comment by luke-jr** [comment]
Which XDG standard defines this?

**Comment by A6GibKm** [comment]
[This one.](https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html)

**Comment by A6GibKm** [comment]
- The Desktop file fails `desktop-file-validate`

```
$ desktop-file-validate org.bitcoin.bitcoin_qt.desktop.in
org.bitcoin.bitcoin_qt.desktop: warning: key "Encoding" in group "Desktop Entry" is deprecated
org.bitcoin.bitcoin_qt.desktop: hint: value "Office;Finance;P2P;Network;Qt;" for key "Categories" in group "Desktop Entry" contains more than one main category; application might appear more than once in the application menu
```

- Consider renaming to org.bitcoincore.bitcoin-qt to match the already existing flatpak at https://github.com/flathub/org.bitcoincore.bitcoin-qt.

**Comment by MarcoFalke** [comment]
This has been needing rebase for about 2 years, so I am closing for now. Can be reopened any time, just let us know when.

**Pull request closed without merging**